src/entities/activity/ui/TimeStatusLabel.tsx

```diff
@@ -0,0 +1,52 @@
+import { ActivityListItem, ActivityStatus } from "../lib"
+
+import { useCustomTranslation } from "~/shared/utils"
+
+interface TimeStatusLabelProps {
+  activity: ActivityListItem
+}
+
+const TimeStatusLabel = ({ activity }: TimeStatusLabelProps) => {
+  const { t } = useCustomTranslation()
+
+  const isStatusScheduled = activity.status === ActivityStatus.Scheduled
+
+  const isStatusPastDue = activity.status === ActivityStatus.PastDue
+
+  const isStatusInProgress = activity.status === ActivityStatus.InProgress
+
+  const hasSceduledAt = isStatusScheduled && activity.hasEventContext && !activity.isTimeoutAllow
+
+  const hasAvailableFromTo = isStatusScheduled && activity.hasEventContext && activity.isTimeoutAllow
+
+  const hasAvailableToOnly = isStatusPastDue
+
+  const hasTimeToComplete =
+    (isStatusScheduled || isStatusPastDue || isStatusInProgress) &&
+    activity.isTimedActivityAllow &&
+    !!activity.timeToComplete
+
+  return (
+    <>
+      {hasSceduledAt && activity.scheduledAt && (
```

Comment:
Please put `activity.scheduledAt` in `hasSceduledAt` variable

Reply:
Done